workflows/benchmarks/quant-comparisons.Rmd

```diff
@@ -0,0 +1,524 @@
+    as("dgCMatrix") # compress sparse matrix
+  dimnames(counts) <- list(readLines(paste0(base,".", "genes.txt")),
+                           readLines(paste0(base,".barcodes.txt")))
+  if(cellfilter == "knee"){
+    # calculate ranks and find knee
+    rank_df <- DropletUtils::barcodeRanks(counts)
+    cells <- which(rank_df$total >= metadata(rank_df)$knee)
+    counts <- counts[, cells]
+  }
+  return(counts)
+}
+```
+
+Make a list of kallisto runs as SCEs
+```{r}
+kallisto_sces <- base_file %>%
+  purrr::map(read_kallisto_counts) %>%
+  purrr::map(~ SingleCellExperiment(list(counts = .x))) 
+
+# add names
+names(kallisto_sces) <- kallisto_quant_ids
+```
+
+Calculate cell QC statistics for each sample
+
+```{r}
+kallisto_sces <- kallisto_sces %>% 
+  purrr::map(
+    ~ scater::addPerCellQC(
+      .x,
+      subsets = list(mito = mito_genes[mito_genes %in% rownames(.x)])
+    ) 
+  )
+```
+
+Make a table of the SCE cell QCs.
+```{r}
+kallisto_cell_qc <- kallisto_sces %>%
+  purrr::map_df(~ as.data.frame(colData(.x)) %>%
+                  tibble::rownames_to_column(var = "cell_id"),
+                .id = "quant_id") %>%
+  dplyr::left_join(quant_info %>% dplyr::filter(tool == "kallisto"), 
+                   by = c("quant_id" = "quant_dir"))
+```
+
+```{r}
+summary(kallisto_cell_qc$sum)
+```
+
+Summarize the number of detected genes per cell.
+```{r}
+summary(kallisto_cell_qc$detected)
+```
+
+### Cellranger
+
+Use `DropletUtils::read10xCounts` to read cellranger counts as SCEs
+
+```{r}
+cellranger_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "cellranger") %>%
+  dplyr::pull(quant_dir)
+
+
+cellranger_sces <- cellranger_quant_ids  %>%
+  purrr::map(
+    ~ read10xCounts(file.path(cellranger_data_dir, .x, 
+                              "outs", "filtered_feature_bc_matrix.h5"),
+                    col.names = TRUE)
+  ) %>% 
+  purrr::map(
+    # get the portion before the "-" which includes the barcode
+    function(x) {
+      colnames(x) <- stringr::str_extract(colnames(x), "^([^-]+)")
+      return(x)
+    }
+  )
+
+# add names
+names(cellranger_sces) <- cellranger_quant_ids
+```
+
+Calculate cell QC statistics for each sample
+
+```{r}
+cellranger_sces <- cellranger_sces %>% 
+  purrr::map(
+    ~ scater::addPerCellQC(
+      .x,
+      subsets = list(mito = mito_genes[mito_genes %in% rownames(.x)])
+    ) 
+  )
+```
+
+Make a table of the SCE cell QCs.
+```{r}
+cellranger_cell_qc <- cellranger_sces %>%
+  purrr::map_df(~ as.data.frame(colData(.x)) %>%
+                  tibble::rownames_to_column(var = "cell_id"),
+                .id = "quant_id") %>%
+  dplyr::left_join(quant_info %>% dplyr::filter(tool == "cellranger"), 
+                   by = c("quant_id" = "quant_dir"))
+```
+
+```{r}
+summary(cellranger_cell_qc$sum)
+```
+
+Summarize the number of detected genes per cell.
+```{r}
+summary(cellranger_cell_qc$detected)
+```
+
+
+
+## Comparing summary stats across tools
```

Comment:
This seems like a good point to break this into it's own notebook. And if it were me, I'd probably write a few functions! 3 tools = do things 3x = time to write function.